### ISSUE TITLE
Add query parameter to filter users by verification state to GET /users API

### ISSUE THREAD

**Issue by isabelcosta** [anitab-org/mentorship-backend#403]
## Description
As a developer,
I need a filter feature on the Users listing endpoint,
so that I can reuse `GET /users` API to filter users who have the email verified.

`GET /users` API needs a way to filter users by email verification state (which is represented by a boolean value). This will allow us deprecating the extra `GET /users/verified` API.

### Update [Required]
- [ ] Add a query parameter support to `GET /users`, to filter by its current state.
- [ ] Update Swagger UI documentation for this API.
- [ ] Write at least one test to prove your new feature is working.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
4 hours

**Comment by Juhi-Purswani** [comment]
@isabelcosta if no one is assigned to this task can I work on it?

**Comment by isabelcosta** [comment]
Yes you can @Juhi-Purswani :)
Be aware that sometimes you may not see someone assigned, but if you see in the comments, someone claiming it and a maintainer approving it, it can mean the contributor is working on it.
No one has claimed this yet, so feel free to work on this and let me know if you have any doubt

**Comment by lorryaze** [comment]
Hi @isabelcosta can I work on this issue?

**Comment by isabelcosta** [comment]
Sorry for the delay in my response @lorryaze I think you are already occupied with other issues, so I will keep this available. Also because there are no updates from @Juhi-Purswani

**Comment by satya7289** [comment]
May I work on this issue @isabelcosta

**Comment by Juhi-Purswani** [comment]
@isabelcosta apologies for the dealy, wasn't able to work due to some reasons. But i have worked on the issue, will submit the PR in a day

**Comment by isabelcosta** [comment]
@Juhi-Purswani are you still working on this? 9 days have passed and you haven't created a PR still. I have no problem with that :) it's better if you update us if you can or cannot work on this than to block the issue for you and not letting other contributors working on this.

We just want some transparency. If you cannot work on this let us know so we can assign it to another contributor.

**Comment by Juhi-Purswani** [comment]
@isabelcosta due to some socio-political reasons there is too much network issue in my area. I may not be able to work for coming 1 week more. Please assign the issue to any other contributor.

**Comment by isabelcosta** [comment]
Thank you so much for the update @Juhi-Purswani . I'm sorry to know about that. I will leave this specific issue available for other contributors then.
Although, I will keep creating issues for first timers so once you are able to contribute you will have issues ;)

**Comment by ashokrayal** [comment]
Hi @isabelcosta,

Can you please assign this to me? I will submit the PR in 1-2 days.
Thanks.

**Comment by ashokrayal** [comment]
> Hi @isabelcosta,
> 
> Can you please assign this to me? I will submit the PR in 1-2 days.
> Thanks.

I was trying to setup the project but I am getting some errors on my system. I will let you know.

**Comment by ashokrayal** [comment]
> > Hi @isabelcosta,
> > Can you please assign this to me? I will submit the PR in 1-2 days.
> > Thanks.
> 
> I was trying to setup the project but I am getting some errors on my system. I will let you know.

I am getting errors for pypiwin32==200
![image](https://user-images.githubusercontent.com/26453959/76858443-787ed480-687d-11ea-9df7-f7d1a2006a2c.png)

**Comment by ashokrayal** [comment]
I am using any random values for the environment variable SECRET_KEY and SECURITY_PASSWORD_SALT and dev for FLASK_ENVIRONMENT_CONFIG. Is this setting ok?

**Comment by ashokrayal** [comment]
> > > Hi @isabelcosta,
> > > Can you please assign this to me? I will submit the PR in 1-2 days.
> > > Thanks.
> > 
> > 
> > I was trying to setup the project but I am getting some errors on my system. I will let you know.
> 
> I am getting errors for pypiwin32==200
> ![image](https://user-images.githubusercontent.com/26453959/76858443-787ed480-687d-11ea-9df7-f7d1a2006a2c.png)

I also tried with docker but :(
![image](https://user-images.githubusercontent.com/26453959/76865621-fd232000-6888-11ea-9e56-9a939a13d87e.png)

**Comment by isabelcosta** [comment]
@Ashokrayal could you check [this topic on Zulip](https://anitab-org.zulipchat.com/#narrow/stream/222534-mentorship-system/topic/Troubleshoot) to see if anyone went through the same issue as you?
Could you also post your problem there, in that way most of the community will be aware and can also help you.

**Comment by aparnavadlamani** [comment]
Hi @isabelcosta. I am new to open source. Can I work on this issue?

**Comment by isabelcosta** [comment]
@Juhi-Purswani are you still interested on working on this issue? Please let me know so that I can know to whom I should assign this.

If you feel you won't be able to work on this, please let us know. I will have to assign this to someone else within the next 24h if you aren't available to work on this. There are plenty of issues available you can work on, when you become available.

**Comment by ashokrayal** [comment]
Hi @isabelcosta,

I was working on this issue. I have also submitted PR.

**Comment by isabelcosta** [comment]
@satya7289 are you still interested in contributing to this PR? If yes please let me know so I can assign it to you.

**Comment by isabelcosta** [comment]
Ok I jsut noticed you are assigned to another issue @satya7289 .
@cs17b005 are you interested in this issue? I can assign this to you if you are.

**Comment by aparnavadlamani** [comment]
Yeah! I am interested. But someone else told they have already submitted a pull request?

**Comment by isabelcosta** [comment]
According to our Community Guidelines, you can only submit a pull request once issue is approved and you are assigned to it. I will assign this to you, as I checked the order of requests to work on this @cs17b005 

@Ashokrayal can you please look into other issues, there are plenty here :)

**Comment by isabelcosta** [comment]
@cs17b005 you can submit an PR whenever you want to fix this :)

**Comment by aparnavadlamani** [comment]
I am working on it. But when I try to run it I get the following error. Could you help me solve it? sqlalchemy.exc.NoSuchModuleError

**Comment by pknotfound** [comment]
Hey @cs17b005 
I am not very good in flask, but I too came through this error when I started with the language.
I referred to [this](https://github.com/miguelgrinberg/Flask-Migrate/issues/197) to solve the issue
Hope this helps.

**Comment by aparnavadlamani** [comment]
I have referred to it and many of them say it was caused due to a typing error in spelling of sqllite but I didn't find that anywhere in this code. @Ashokrayal could you please tell me how you solved the errors you got in your machine and also can you help me on this one?

**Comment by Juhi-Purswani** [comment]
@isabelcosta Yes I am still interested and I will be available to work but since this issue has already been assigned I will search for some other issues. Thanks.

**Comment by isabelcosta** [comment]
@Juhi-Purswani oh ok! I am sure we have plenty of issues for you to work on, let me know if you need any help.

**Comment by ashokrayal** [comment]
@cs17b005 I have commented few packages in requirement file which were related to docker. I found this solution on tulip app.

**Comment by isabelcosta** [comment]
@cs17b005 are you still in need of help for this issue?

**Comment by aparnavadlamani** [comment]
Yes I am still unable to find the issue even on Zulip.

**Comment by isabelcosta** [comment]
@cs17b005 Could you please share your issue again, here or on Zulip (tagging me there)?

**Comment by aparnavadlamani** [comment]
Actually it's the same issue which @Ashokrayal posted above and as he suggested when I removed some packages related to docker the app was up and running. Thank you for the support! I'll complete it as soon as possible. Will let you know if I come across anything else. 
The changes I made to the requirements doc are:
I upgraded the MarkupSafe package and also I removed the docker related packages from the requirements file. It works perfectly!

**Comment by isabelcosta** [comment]
@cs17b005 yey! glad you solved the problem you were having :D

**Comment by aparnavadlamani** [comment]
I have made few changes but when I try to register a user it gives me Internal Server error. I have searched for this issues in the Issues already posted and found related issues but I didn't find a solution. I have tried one of the solutions which says enable 2-step verification and then generate a password and use the generated password but this doesn't work for me either. Could you help me with this? @isabelcosta

**Comment by isabelcosta** [comment]
@cs17b005 I am sorry for the delay in my response, could you tell me if you still have your a problem with the development environment setup.

**Comment by aparnavadlamani** [comment]
Yes, I am still facing the issue. No problem you might have been busy. I have seen similar issues with some other user but I couldn't get any help from that post.

**Comment by SanketDG** [comment]
@cs17b005 Can you post the issue on Zulip and tag me? I might be able to help you here with the dev setup.

Please paste the complete error messages using Github's gists and then paste the link there. Also list down the exact steps that you followed from the setup guide.

**Comment by vj-codes** [comment]
Unassigning @cs17b005 due to inactivity.

**Comment by ashokrayal** [comment]
Hi @vj-codes,

Can you please assign this issue to me? 
I have worked on it in past. I also submitted a PR #509 . But the issue got assigned to someone else in between.

**Comment by vj-codes** [comment]
@Ashokrayal hello you made have your first contribution to this repo through #848 and since this is a` first timers only issue`, you are no longer eligible

@isabelcosta  what is the priority of this issue, should I add the OSH label for newcomers to work in this?
And if not for OSH , can @Ashokrayal be assigned since no one else is claiming.Thanks!

**Comment by isabelcosta** [comment]
This is an important feature. If @Ashokrayal is available and interested in working on this then it's okay to assign it to them, in case @Ashokrayal is not already assigned to an issue

**Comment by ashokrayal** [comment]
Hi @vj-codes, 
I am available. Please assign it to me :)

**Comment by vj-codes** [comment]
Assigning you @Ashokrayal 
Happy coding!

**Comment by vj-codes** [comment]
@Ashokrayal how's the PR going?

**Comment by ashokrayal** [comment]
Hi @vj-codes,
I will submit the PR today.
Thanks :)

**Comment by NenadPantelic** [comment]
HI @isabelcosta, can I take this one. Also, I have a suggestion, I put that in one of the comments of the PR that's closely related to this. I suggested more universal search functionality - multiparams search, by verification state, location, name, skills, etc. Of course, that can be created as a completely new issue with subtasks. Here's the link of that comment: https://github.com/anitab-org/mentorship-backend/pull/488#issuecomment-762801282
